internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/probe.go

```diff
@@ -107,38 +144,97 @@ func New(logger *slog.Logger) probe.Probe {
 				tracerIDContainsSchemaURL{},
 				tracerIDContainsScopeAttributes{},
 			},
-			Uprobes: []probe.Uprobe{
+			Uprobes: []*probe.Uprobe{
+				uprobeNewStart,
 				{
 					Sym:         "go.opentelemetry.io/otel/internal/global.(*tracer).Start",
 					EntryProbe:  "uprobe_Start",
 					ReturnProbe: "uprobe_Start_Returns",
+					PackageConstrainsts: []probe.PackageConstrainst{
+						otelWithoutAutoSDK,
+					},
 				},
 				{
 					Sym:        "go.opentelemetry.io/otel/internal/global.(*nonRecordingSpan).End",
 					EntryProbe: "uprobe_End",
+					PackageConstrainsts: []probe.PackageConstrainst{
+						otelWithoutAutoSDK,
+					},
 				},
 				{
 					Sym:         "go.opentelemetry.io/otel/internal/global.(*nonRecordingSpan).SetAttributes",
 					EntryProbe:  "uprobe_SetAttributes",
 					FailureMode: probe.FailureModeIgnore,
+					PackageConstrainsts: []probe.PackageConstrainst{
+						otelWithoutAutoSDK,
+					},
 				},
 				{
 					Sym:         "go.opentelemetry.io/otel/internal/global.(*nonRecordingSpan).SetStatus",
 					EntryProbe:  "uprobe_SetStatus",
 					FailureMode: probe.FailureModeIgnore,
+					PackageConstrainsts: []probe.PackageConstrainst{
+						otelWithoutAutoSDK,
+					},
 				},
 				{
 					Sym:         "go.opentelemetry.io/otel/internal/global.(*nonRecordingSpan).SetName",
 					EntryProbe:  "uprobe_SetName",
 					FailureMode: probe.FailureModeIgnore,
+					PackageConstrainsts: []probe.PackageConstrainst{
+						otelWithoutAutoSDK,
+					},
 				},
 			},
-			SpecFn: loadBpf,
+			SpecFn:        loadBpf,
+			ProcessRecord: c.decodeEvent,
 		},
 		ProcessFn: processFn,
 	}
 }
 
+type recordKind uint32
+
+const (
+	recordKindTelemetry recordKind = iota
+	recordKindConrol
+)
+
+type converter struct {
+	logger *slog.Logger
+
+	uprobeNewStart   *probe.Uprobe
+	uprobeNewStartMu sync.Mutex
+}
+
+func (c *converter) decodeEvent(record perf.Record) (*event, error) {
+	reader := bytes.NewReader(record.RawSample)
+
+	var kind recordKind
+	err := binary.Read(reader, binary.LittleEndian, &kind)
+	if err != nil {
+		return nil, err
+	}
+
+	var e *event
+	switch kind {
+	case recordKindTelemetry:
+		e = new(event)
+		reader.Reset(record.RawSample)
+		err = binary.Read(reader, binary.LittleEndian, e)
+	case recordKindConrol:
+		c.uprobeNewStartMu.Lock()
```

Comment:
The `decodeEvent`  function of a probe instance can't be called from multiple goroutines , right? If so, maybe should remove the mutex